monailabel/utils/others/init_direction_and_orient.py

```diff
@@ -0,0 +1,72 @@
+import os
+import pathlib as Path
+from typing import List, Union
+
+import SimpleITK as sitk
+
+
+def get_file_paths(
+    folder: Union[str, Path], extension_tuple=(".nrrd"), pattern: str = None, ignore_folders: Union[List, str] = []
+) -> Union[List[str], List[Path]]:
+    """
+    Given a directory return all image paths within all sibdirectories.
+
+    Parameters
+    ----------
+    folder
+        Where to look for images
+    extension_tuple
+        Select only images with these extensions
+    pattern
+        Do a simple `pattern in filename` filter on filenames
+    ignore_folders
+        do not look in folder with these names
+
+    Notes
+    -----
+    Lama is currently using a mixture of Paths or str to represent filepaths. Will move all to Path.
+    For now, return same type as folder input
+
+    Do not include hidden filenames
+    """
+    paths = []
+
+    if isinstance(ignore_folders, str):
+        ignore_folders = [ignore_folders]
+
+    for root, subfolders, files in os.walk(folder):
+        for f in ignore_folders:
+            if f in subfolders:
+                subfolders.remove(f)
+
+        for filename in files:
+            if filename.lower().endswith(extension_tuple) and not filename.startswith("."):
+                if pattern:
+                    if pattern and pattern not in filename:
+                        continue
+
+                paths.append(os.path.abspath(os.path.join(root, filename)))
+
+    if isinstance(folder, str):
+        return paths
+    else:
+        return [Path(x) for x in paths]
+
+
+def initialise_ori_and_info(target_dir):
+    """
+    output from MPI2/LAMA removes direction and orientation information, leaving them to be blank within headers.
+    This function should initialise direction and orientation information
+    """
+    volpaths = get_file_paths(target_dir)
+
+    for i, path in enumerate(volpaths):
+        vol = sitk.ReadImage(path)
+        # pa = sitk.PermuteAxesImageFilter()
+        # pa.SetOrder([1,0,2])
+        # flipped_vol = pa.Execute(vol)
+
+        # sets Direction and Origin
+        vol.SetDirection([-1.0, 0.0, 0.0, 0.0, -1.0, 0.0, 0.0, 0.0, 1.0])
```

Comment:
Note that spacing is included in 'space directions` in the NRRD file header but in SimpleITK images there is a separate `SetSpacing` method for it.

Reply:
Right, because we should have the spacing in our output. 

If needed, we can just add vol.SetSpacing([1.0, 1.0, 1.0]) as line 71 and that should do the trick? 
Let me know your opinions.

Reply:
SimpleITK writer will always write the image geometry (origin, spacing, axis directions) to the output image file, I just wanted to mention this to make sure that no information is lost when this code is used in a different context.

The main issue though is that the image geometry should not be set arbitrarily. For example, how do we know that the IJK image axes are RAS (encoded by the -1 values in the direction matrix) and not LPS (direction matrix is identity matrix)? Instead of setting some hardcoded default geometry, it should be copied from the input image.